gzipi/lib.py

```diff
@@ -326,15 +457,17 @@ def __next__(self):
             return self.current_line
 
 
-def retrieve(keys_fin, file_path, index_fin, output_stream):
+def retrieve(
+    keys_fin: IO[bytes], file_path: str, index_fin: IO[str], output_stream: IO[bytes]
+) -> None:
     """Retrieve data from an indexed file.
 
-    :param file keys_fin: A steam with list of keys to retrieve.
-    :param str file_path: A local S3 path to the file retrieve data from.
-    :param str index_fin: A file stream to read index from.
-    :param file output_stream: A file stream to output results to.
+    :param keys_fin: A steam with list of keys to retrieve.
+    :param file_path: A local S3 path to the file retrieve data from.
+    :param index_fin: A file stream to read index from.
+    :param output_stream: A file stream to output results to.
     """
-
+    compression = _get_compression_type(file_path)
```

Comment:
Do we need _get_compression_type as a separate function? We already have _getermine_compression_from_header, why not use that here directly instead? We have the open file object on the line below.